packages/funding-rate-proposer/src/proposer.js

```diff
@@ -163,8 +163,33 @@ class FundingRateProposer {
 
     // Assume pricefeed has been cached and updated prior to this function via the `update()` call.
     const priceFeed = this.priceFeedCache[fundingRateIdentifier];
-    const pricefeedPrice = await this._getPriceFeedAndPrice(fundingRateIdentifier);
-    if (!pricefeedPrice) return;
+    if (!priceFeed) {
+      this.logger.error({
+        at: "PerpetualProposer",
+        message: "Failed to create pricefeed for funding rate identifier",
+        fundingRateIdentifier
+      });
+      return null;
+    }
+    // Get hypothetical timestamp to use for a new price proposal:
+    // Unless `usePriceFeedTime=true`, use the latest block's timestamp as the request
+    // timestamp so that the contract does not interpret `requestTimestamp` as being in the future.
+    const requestTimestamp = usePriceFeedTime
+      ? priceFeed.getLastUpdateTime()
+      : (await this.web3.eth.getBlock("latest")).timestamp;
```

Comment:
Is it a problem if the latest block time is > the price feed's latest update time?

Reply:
Or we can do like the `optimistic-oracle-proposer` and just conservatively `update()` the priceFeed before fetching historical price. [reference](https://github.com/UMAprotocol/protocol/blob/master/packages/optimistic-oracle/src/proposer.js#L262)

Reply:
I implemented the above ^

Reply:
Sounds good! This isn't guaranteed to have an update time > block time, but I think this is probably as close as we can reasonably get:
1. Price feeds usually throttle to ensure they don't update more than once per minute by default.
2. Block times can (but usually don't) be ahead of wall-clock time (they have a 90 second or so tolerance if I remember correctly).

Reply:
yeah the other solution which I think is a bit over engineered for now is something like

```js
let requestTimestamp;

// Update pricefeed until its last update time is caught up with the latest network block, or
// break early after 1 minute and assume that's good enough (or throw).
setTimeout(function() {
    while(true) {
        await priceFeed.update();
        requestTimestamp = (await web3.eth.getBlock()).timestamp
        if (priceFeed.lastUpdateTime() >= requestTimestamp) break;
    }
},
60000
```

Reply:
Yeah, agree with your decision to do it the way you have it in the PR. LGTM!

Reply:
But this wouldn't be useful unless the `minTimeBetweenUpdates` was respected, and given the expected iteration cadence of 5 minutes we don't have much timeout period to play with. This sort of method would make sense in a looping infrastructure